src/gui/sdlmain.cpp

```diff
@@ -4483,6 +4383,10 @@ void config_add_sdl() {
 	        "  <custom>:  Limit the viewport within to a custom resolution or percentage of\n"
 	        "             the desktop. Specified in WxH, N%, N.M%. Examples: 960x720 or 50%");
 
+	pstring = sdl_sec->Add_string("integer_scale_mode", always, "off");
+	pstring->Set_help(
+	        "Scale output by integer factor along set axis: off, horizontal, vertical.");
+
```

Comment:
I suggest to rename this to `integer_scaling` and the following description that follows our new, normalised description style:

```
Constrain the horizontal or vertical scaling factor to integer values. The
correct aspect ratio is always maintained, which may result in a non-integer
scaling factor to be applied in the other dimension.
  off:         No integer scaling constraint; the aspect ratio correct image
               fills the viewport (default).
  horizontal:  Constrain the horizontal scaling factor to integer values
               within the viewport.
  vertical:    Constrain the vertical scaling factor to integer values
               within the viewport.

```

Reply:
Also, `aspect` is in `[render]`, so I think this belongs to the `[render]` section too. I don't think we should add anything new to `[sdl]`, and in fact `viewport_resolution` should be moved from `[sdl]` to `[render]` as well. 

The whole `[sdl`] section is quite ill-named in itself. Why are implementation details leaking into the config? No one cares if we use SDL, GLFW, Unity, framework XYZ, or no framework at all—the configuration should be completely agnostic of that _implementation_ detail... We should move things _out_ from there into more appropriately named sections, like @FeralChild64 did with the `[mouse]` section—definitely we should not add to the mess.